tox.ini

```diff
@@ -0,0 +1,43 @@
+[flake8]
+max-line-length = 88
+max-complexity = 10
+inline-quotes = double
+
+
+[tox]
+envlist = py27,py36
+
+
+[testenv]
+setenv = AWS_DEFAULT_REGION = us-east-1
+extras = test
+commands =
+    python -m pytest --cov watchbot_progress --cov-report term-missing --ignore=venv
+
+
+[testenv:flake8]
+basepython = python3
+skip_install = true
+deps =
+    flake8
+commands =
+    flake8 watchbot_progress
+
+
+[testenv:pylint]
+basepython = python3
+skip_install = true
+deps =
+    pyflakes
+    pylint
+commands =
+    pylint watchbot_progress
+
+
+[testenv:doc]
+select = D1
+ignore = D105
+deps =
+    pydocstyle
+commands =
+    python -m pydocstyle watchbot_progress
```

Comment:
Let's keep linting separate from testing. We should remove flake8, pydocstyle and pylint from the tox config and use `pre-commit` hooks. (for example, see https://github.com/mapbox/rio-merge-rgba/pull/11)

Reply:
@perrygeo Can I keep the first `[flake8]` block which can be use by editors ?

Reply:
Does this mean flake8 will get run twice, once under pre-commit , once under tox? Not worried about the extra runtime but it could be problematic to have duplicate configs when we forget to keep the two in configurations in sync. I'd prefer to have just a single point of configuration in pre-commit.

Reply:
@perrygeo by default the `tox` command will run only the `[testenv]` commands defined with https://github.com/mapbox/watchbot-progress-py/pull/25/files#diff-b91f3d5bd63fcd17221b267e851608e8R8 (so here py27 and py36) to run other `env` we need to set `TOXENV` variable e.g: `TOXENV=flake8 tox (this will run the `[testenv:flake8]` block.
The first `[flake8]` block is just here to be a global flake8 configuration.

is that clearer ?

Reply:
👌